src/facade/dragonfly_connection.cc

```diff
@@ -1333,4 +1343,26 @@ Connection::MemoryUsage Connection::GetMemoryUsage() const {
   };
 }
 
+Connection::WeakRef::WeakRef(std::shared_ptr<Connection> ptr, QueueBackpressure* backpressure,
+                             unsigned thread)
+    : ptr_{ptr}, backpressure_{backpressure}, thread_{thread} {
+}
+
+unsigned Connection::WeakRef::Thread() const {
+  return thread_;
+}
+
+Connection* Connection::WeakRef::Get() const {
+  DCHECK_EQ(ProactorBase::me()->GetPoolIndex(), int(thread_));
```

Comment:
Any reason to have `thread_` of type `unsigned` instead of `int`? It looks like it's both assigned from `GetPoolIndex()` and checked against it..

Reply:
Because the only reason it's signed is to use -1 as an invalid value 😭 I can change it, but this code isn't supposed to contain invalid ids anywhere

Reply:
yeah, we should follow up on your suiggestion and change it in helio to uint_max.